runbot_gitlab_minimal/models/runbot_repo.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# Copyright <2017> <Vauxoo info@vauxoo.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import os
+import logging
+import re
+import urllib
+
+import requests
+
+from openerp import models, fields
+
+_logger = logging.getLogger(__name__)
+
+
+class RunbotRepo(models.Model):
+    _inherit = "runbot.repo"
+
+    uses_gitlab = fields.Boolean(help='Enable the ability to use gitlab '
+                                      'instead of github')
+
+    def git(self, cr, uid, ids, cmd, context=None):
+        """Rewriting the parent method to avoid deleting the merge_request the
+        gitlab"""
+        if cmd == ['fetch', '-p', 'origin', '+refs/pull/*/head:refs/pull/*']:
+            cmd.remove('-p')
+        return super(RunbotRepo, self).git(cr, uid, ids, cmd, context=context)
+
+    def update_git(self, cr, uid, repo, context=None):
+        """Download the gitlab merge request references to work with the
+        referrals of pull github"""
+        if os.path.isdir(os.path.join(repo.path, 'refs')) and repo.uses_gitlab:
+            repo.git(['fetch', '-p', 'origin',
+                      '+refs/merge-requests/*/head:refs/pull/*'])
+        return super(RunbotRepo, self).update_git(cr, uid, repo,
+                                                  context=context)
+
+    def github(self, cr, uid, ids, url, payload=None, ignore_errors=False,
+               context=None):
+        """Translate each request from github format to gitlab format"""
```

Comment:
Add to docstring the supported cases:
1. Get info from merge requests
  - input: URL_GITLAB/projects/... instead of URL_GITHUB/repos/
  - output: gitlab res['base']['ref'] = res['gitlab_base_mr'], gitlab res['state'] = res['gitlab_state'].replace('gitlab_state', 'github_state'), gitlab ['head']['ref'] = res['gitlab_head_mr']
2. Statuses
  - input: URL_GITLABL/... instead of URL_GITHUB/statuses/...
  - output: N/A